### ISSUE TITLE
nonportable_path_linter lints gsub("[^0-9\\-]", "", x)

### ISSUE THREAD

**Issue by mcol** [r-lib/lintr#2949]
This shows the problem:
```R
lintr::lint(text = 'gsub("[^0-9\\\\-]", "", x)', linter = lintr::nonportable_path_linter())
# <text>:1:7: warning: [nonportable_path_linter] Use file.path() to construct portable file paths.
# gsub("[^0-9\\-]", "", x)
#       ^~~~~~~~~
```

A minimized version:
```R
lintr::lint(text = "'[\\\\-]'", linter = lintr::nonportable_path_linter())
# <text>:1:2: warning: [nonportable_path_linter] Use file.path() to construct portable file paths.
# '[\\-]'
#  ^~~~~
```

---

Actually, my script has `gsub("[^0-9\\-]", "", x)`, but this is parsed differently when using `text`:
```R
lintr::lint(text = "gsub('[^0-9\\-]', '', x)", linter = lintr::nonportable_path_linter())
# <text>:1:13: error: [error] '\-' is an unrecognized escape in character string.
# gsub('[^0-9\-]', '', x)
```
